docs/content/docs/connectors/datastream/elasticsearch.md

```diff
@@ -65,240 +61,90 @@ about how to package the program with the libraries for cluster execution.
+        .type("my-type")
+        .id(element)
+        .source(json);
+}
 ```
 {{< /tab >}}
 {{< tab "scala, Elasticsearch 6.x and above" >}}
 ```scala
-import org.apache.flink.api.common.functions.RuntimeContext
+import org.apache.flink.api.connector.sink.SinkWriter
+import org.apache.flink.connector.elasticsearch.sink.{Elasticsearch6SinkBuilder, RequestIndexer}
 import org.apache.flink.streaming.api.datastream.DataStream
-import org.apache.flink.streaming.connectors.elasticsearch.ElasticsearchSinkFunction
-import org.apache.flink.streaming.connectors.elasticsearch.RequestIndexer
-import org.apache.flink.streaming.connectors.elasticsearch6.ElasticsearchSink
-
 import org.apache.http.HttpHost
 import org.elasticsearch.action.index.IndexRequest
 import org.elasticsearch.client.Requests
 
-import java.util.ArrayList
-import java.util.List
+...
 
 val input: DataStream[String] = ...
 
-val httpHosts = new java.util.ArrayList[HttpHost]
-httpHosts.add(new HttpHost("127.0.0.1", 9200, "http"))
-httpHosts.add(new HttpHost("10.2.3.1", 9200, "http"))
-
-val esSinkBuilder = new ElasticsearchSink.Builder[String](
-  httpHosts,
-  new ElasticsearchSinkFunction[String] {
-     def process(element: String, ctx: RuntimeContext, indexer: RequestIndexer) {
-          val json = new java.util.HashMap[String, String]
-          json.put("data", element)
-
-          val rqst: IndexRequest = Requests.indexRequest
-            .index("my-index")
-            .`type`("my-type")
-            .source(json)
-
-          indexer.add(rqst)
-     } 
-  }
-)
-
-// configuration for the bulk requests; this instructs the sink to emit after every element, otherwise they would be buffered
-esSinkBuilder.setBulkFlushMaxActions(1)
-
-// provide a RestClientFactory for custom configuration on the internally created REST client
-esSinkBuilder.setRestClientFactory(new RestClientFactory {
-  override def configureRestClientBuilder(restClientBuilder: RestClientBuilder): Unit = {
-       restClientBuilder.setDefaultHeaders(...)
-       restClientBuilder.setMaxRetryTimeoutMillis(...)
-       restClientBuilder.setPathPrefix(...)
-       restClientBuilder.setHttpClientConfigCallback(...)
-  }
-})
-
-// finally, build and add the sink to the job's pipeline
-input.addSink(esSinkBuilder.build)
+input.sinkTo(
+  new Elasticsearch6SinkBuilder[String] // For Elasticsearch 7.x use Elasticsearch7SinkBuilder
+    .setBulkFlushMaxActions(1) // Instructs the sink to emit after every element, otherwise they would be buffered
+    .setHosts(new HttpHost("127.0.0.1", 9200, "http"))
+    .setEmitter((element: String, context: SinkWriter.Context, indexer: RequestIndexer) => 
+    indexer.add(createIndexRequest(element)))
+    .build())
+
+...
+
+def createIndexRequest(element: (String)): IndexRequest = {
+
+  val json = Map(
+    "data" -> element.asInstanceOf[AnyRef]
+  )
+
+  Requests.indexRequest.index("my-index").`type`("my-type").source(mapAsJavaMap(json))
+}
 ```
 {{< /tab >}}
```

Comment:
I think we should also add a tab for ES 7 and maybe simplify the tab structure i.e. only have two tabs either java or scala and two code blocks within a tab using ES6 and ES7

Reply:
I had a look at the other connectors and kept the tabs as java and scala and added an example for each version in there